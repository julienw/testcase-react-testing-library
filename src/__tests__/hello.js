import "react-testing-library/cleanup-after-each";
import React from "react";
import { render, fireEvent } from "react-testing-library";
import Hello from "../hello";

test("It properly renders the component", () => {
  const { container } = render(<Hello />);
  fireEvent.mouseMove(container.querySelector('.playground'), {
    clientX: 5,
    clientY: 10,
  });

  expect(container.firstChild).toMatchSnapshot();

  fireEvent.mouseMove(container.querySelector('.playground'), {
    clientX: 5,
    clientY: 10,
    pageX: 5,
    pageX: 10,
    offsetX: 2,
    offsetY: 7,
  });

  expect(container.firstChild).toMatchSnapshot();
});
