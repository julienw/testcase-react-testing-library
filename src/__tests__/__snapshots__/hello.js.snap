// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It properly renders the component 1`] = `
<div
  class="playground"
  style="width: 500px; height: 500px; white-space: pre;"
>
  clientX: 5
clientY: 10
pageX: undefined
pageY: undefined
x: undefined
y: undefined
offsetX: undefined
offsetY: undefined
</div>
`;

exports[`It properly renders the component 2`] = `
<div
  class="playground"
  style="width: 500px; height: 500px; white-space: pre;"
>
  clientX: 5
clientY: 10
pageX: undefined
pageY: undefined
x: undefined
y: undefined
offsetX: undefined
offsetY: undefined
</div>
`;
